examples/jupyter-tutorials/API.ipynb

```diff
@@ -11,188 +11,1039 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Expression building\n",
+    "### Expression Building\n",
     "\n",
-    "(note: may have old API in some cases)"
+    "(This tutorial is tested on DyNet 2.0.3+ and Python 2.7.)\n",
+    "\n",
+    "If you find any issue while running and you think you are right, please try to restart the kernel and run it again. :)"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {
-    "collapsed": false
-   },
+   "metadata": {},
    "outputs": [],
    "source": [
+    "from __future__ import print_function\n",
+    "from __future__ import division\n",
+    "from __future__ import absolute_import\n",
+    "\n",
+    "# Note: please import dynet_config before import dynet\n",
+    "import dynet_config\n",
+    "# set random seed to have the same result each time\n",
+    "dynet_config.set(random_seed=0)\n",
     "import dynet as dy\n",
     "\n",
+    "import numpy as np\n",
+    "\n",
     "## ==== Create a new computation graph\n",
     "# (it is a singleton, we have one at each stage.\n",
-    "# dy.renew_cg() clears the current one and starts anew)\n",
-    "dy.renew_cg()\n",
+    "# dy.renew_cg() clears the current one and starts a new one)\n",
+    "dy.renew_cg();"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create Expressions\n",
+    "Expressions are used as an interface to the various functions that can be used to build DyNet computation graphs. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create a scalar expression.\n",
+    "value = 5.0\n",
+    "x = dy.scalarInput(value)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create a vector expression.\n",
+    "dimension = 3\n",
+    "v = dy.vecInput(dimension)\n",
+    "v.set([1,2,3])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create a matrix expression from list\n",
+    "mat1 = dy.inputTensor([[1,2], [3,4]]) # Row major\n",
     "\n",
-    "## ==== Creating Expressions from user input / constants.\n",
-    "x = dy.scalarInput(value)\n",
+    "# or, using numpy array\n",
+    "mat2 = dy.inputTensor(np.array([[1,2], [3,4]]))\n",
     "\n",
-    "v = dy.vecInput(dimension)\n",
-    "v.set([1,2,3])\n",
+    "mat3 = dy.inputTensor(np.zeros((2,3)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create a vector/matrix expression of special values\n",
+    "# Different from TensorFlow, DyNet treats batch as the last dim\n",
```

Comment:
This is not exactly correct. DyNet has a special "batch" dimension, so that you don't need to think about the batch dimension most of the time other than during input and output. Could we make this a little more accurate, following the explanation here?:
http://dynet.readthedocs.io/en/latest/minibatch.html